### ISSUE TITLE
Clarification on Epoch Number for Reproducing Table 2 Results

### ISSUE THREAD

**Issue by X1AOX1A** [PRIME-RL/Entropy-Mechanism-of-RL#4]
Hi, thanks for sharing this work — it's been very helpful.

I’m trying to reproduce the results in Table 2 and had a quick question about the training configuration. In the script [7b_kl_cov.sh](https://github.com/PRIME-RL/Entropy-Mechanism-of-RL/blob/5c8555e58859ab0c5f5b2a7589c009b304caf389/recipe/dapo/7b_kl_cov.sh#L139), the number of training epochs is set to 1000.

https://github.com/PRIME-RL/Entropy-Mechanism-of-RL/blob/5c8555e58859ab0c5f5b2a7589c009b304caf389/recipe/dapo/7b_kl_cov.sh#L139

Could you please confirm whether this is the same epoch number used to produce the results in Table 2? It seems quite high for a typical training setup. Or was a different number used during the actual experiments?

Thanks in advance!

**Comment by Raf-Chen** [comment]
The epoch number means how many times of the training dataset are used, the step in our paper is gradient step and they are different. You do not need to modify this number as you can stop training at any time.

**Comment by X1AOX1A** [comment]
Thanks for your reply.

Sure, one can stop at any time, but an overly long epoch number could affect the learning rate schedule (as it typically depends on the total number of epochs).

Also, could you kindly provide the exact number of steps or epochs used to reproduce the results in Table 2?

**Comment by Raf-Chen** [comment]
In fact, as presented in our paper, we keep learning rate  constant, and as I said before, we align the gradient step. We do not record how many “epochs” of data are used. But I guess for your 7B model, set epoch as 15-20 might be a good choice.

**Comment by X1AOX1A** [comment]
Thanks for the prompt and helpful reply — really appreciate your support!
